feat: Add SearchBar component to Nav

techwithanirudh opened the pull request:
This commit adds the SearchBar component to the navigation bar in order to enable search functionality. Now users can easily search for specific items within the navigation.

﻿Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.



## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation update  
  

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration:
##


### **Test Configuration**:
##


## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

techwithanirudh committed:
feat: Add SearchBar component to Nav

This commit adds the SearchBar component to the navigation bar in order to enable search functionality. Now users can easily search for specific items within the navigation.

techwithanirudh committed:
Refactor Nav and SearchBar components

The commit refactors the Nav component by moving the SearchBar component within the Nav component. This change ensures that the SearchBar is rendered only when the isSearchEnabled condition is true.

In addition, the commit also modifies the styling of the SearchBar component by adding rounded corners and border to enhance the visual appearance.

techwithanirudh committed:
Update gitignore

techwithanirudh committed:
C

Refactor search bar styles

This commit refactors the styles of the search bar component in the Nav component. The border color and hover background color have been modified to improve the visual appearance.

techwithanirudh committed:
Fix margin

danny-avila commented on client/src/components/Nav/Logout.jsx:
```diff
@@ -20,8 +20,7 @@ const Logout = forwardRef(() => {
       <LogOutIcon />
-      {user?.username || localize(lang, 'com_nav_user')}
-      <small>{localize(lang, 'com_nav_log_out')}</small>
+      {localize(lang, 'com_nav_log_out')}
```
refactor this component to TSX and utilize useLocalize hook. CTRL+F for examples, there are many.

techwithanirudh replied:
@danny-avila Fixed!

danny-avila reviewed (changes requested):
Also, run `npm run lint` and make sure you didn't introduce any errors.

techwithanirudh committed:
Rename Logout.jsx to Logout.tsx and update import statements accordingly.
Replace the use of Recoil and store with useLocalize hook for localization.
Update the usage of localize function by removing the lang parameter.
